server/text_generation_server/utils/weights.py

```diff
@@ -762,6 +767,50 @@ def _set_gptq_params(self, model_id, revision):
                 except Exception:
                     pass
 
+    def get_kv_cache_scaling_factor(
+        self, prefix: str, kv_cache_dtype: str, kv_cache_torch_dtype: str
+    ):
+        try:
+            kv_scale = self.get_tensor(f"{prefix}.kv_scale").cpu().tolist()
+        except RuntimeError:
+            if kv_cache_dtype == "fp8":
+                log_once(
+                    logger.warning,
+                    "Could not find the `kv_scale` in checkpoint for `fp8_e4m3`. Using scaling factor"
+                    "`1.0`. This may result in accuracy issues. Please ensure the checkpoint includes "
+                    "the correct KV cache scaling factor.",
+                )
+
+            kv_scale = 1.0
+        else:
+            if kv_cache_dtype == "fp8_e5m2":
+                raise RuntimeError(
+                    "Found `kv_scale` in the checkpoint, but `fp8_e5m2` KV dtype do not support `kv_scale` > 1.0"
+                )
+
+            if not isinstance(kv_scale, float):
+                raise RuntimeError(
+                    "Only support per-tensor scaling factor for `fp8 (fp8_e4m3)` KV cache"
+                )
+
+            if kv_cache_torch_dtype not in {"float8_e4m3fn", "float8_e4m3fnuz"}:
+                raise RuntimeError(
+                    f"Found `kv_scale` in the checkpoint, the config must specify the `kv_cache_torch_dtype` "
+                    f"used for generating kv scales. Expected 'float8_e4m3fn' or 'float8_e4m3fnuz', but got '{kv_cache_torch_dtype}'."
+                )
+
+            # ROCm uses FP8 format with fp8_e4m3fn, whereas Nvidia GPUs use fp8_e4m3.
```

Comment:
Is it should be "ROCm uses FP8 format with fp8_e4m3fnuz, whereas NVIDIA GPU uses fp8_e4m3fn"?

Reply:
Yes that's correct